app/uk/gov/hmrc/tariffclassificationfrontend/service/FileStoreService.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tariffclassificationfrontend.service
+
+import javax.inject.{Inject, Singleton}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.tariffclassificationfrontend.connector.FileStoreConnector
+import uk.gov.hmrc.tariffclassificationfrontend.models._
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+@Singleton
+class FileStoreService @Inject()(connector: FileStoreConnector) {
+
+  def getAttachments(c: Case)(implicit hc: HeaderCarrier): Future[Seq[StoredAttachment]] = {
+    val attachmentsById: Map[String, Attachment] = c.attachments.map(a => a.id -> a).toMap
+    connector.get(c.attachments) map { files =>
+      files map { file =>
+        attachmentsById
+          .get(file.id)
+          .map(StoredAttachment(_, file))
+      } filter(_.isDefined) map (_.get)
+    }
+  }
+
+  def getLetterOfAuthority(c: Case)(implicit hc: HeaderCarrier): Future[Option[StoredAttachment]] = {
+    if(c.application.isBTI) {
+      c.application.asBTI.agent match {
+        case Some(agent: AgentDetails) =>
+          connector
+            .get(agent.letterOfAuthorisation)
+            .map(_.map(StoredAttachment(agent.letterOfAuthorisation, _)))
```

Comment:
As on the other PR, this is a case where an agent exist and the letterOfAuth should be there, what we could do if the Option[] comes empty? 
Maybe this is a case where we just have NO letter... but it looks more like we have a Letter (attatchment) but we could not found it in the filestore.

Reply:
I think from a UI perspective "having no letter" and "having a letter that doesn't exist" means the same thing (the operator has to contact the agent to ask for it)

Reply:
I completely agree but maybe a warning that a required file hasn't been found...